api-catalog-services/src/main/java/org/zowe/apiml/apicatalog/security/SecurityConfiguration.java

```diff
@@ -44,35 +53,97 @@
     private final GatewayTokenProvider gatewayTokenProvider;
 
-    @Override
-    protected void configure(AuthenticationManagerBuilder auth) {
-        auth.authenticationProvider(gatewayLoginProvider);
-        auth.authenticationProvider(gatewayTokenProvider);
+    /**
+     * Filter chain for protecting /apidoc/** endpoints with MF credentials for client certificate.
+     */
+    @Configuration
+    @Order(1)
+    public class FilterChainBasicAuthOrTokenOrCertForApiDoc extends WebSecurityConfigurerAdapter {
+
+        @Value("${apiml.security.ssl.verifySslCertificatesOfServices:true}")
+        private boolean verifySslCertificatesOfServices;
+
+        @Value("${apiml.security.ssl.nonStrictVerifySslCertificatesOfServices:false}")
+        private boolean nonStrictVerifySslCertificatesOfServices;
+
+        @Value("${server.attls.enabled:false}")
+        private boolean isAttlsEnabled;
+
+        @Override
+        protected void configure(AuthenticationManagerBuilder auth) {
+            auth.authenticationProvider(gatewayLoginProvider);
+            auth.authenticationProvider(gatewayTokenProvider);
+        }
+
+        @Override
+        public void configure(WebSecurity web) {
+            configureNoSecurityEndpoints(web);
+        }
+
+        @Override
+        protected void configure(HttpSecurity http) throws Exception {
+            mainframeCredentialsConfiguration(
+                baseConfiguration(http.antMatcher(APIDOC_ROUTES)),
+                authenticationManager()
+            )
+                .authorizeRequests()
+                .antMatchers(APIDOC_ROUTES).authenticated();
+
+            if (verifySslCertificatesOfServices || nonStrictVerifySslCertificatesOfServices) {
+                http.x509().userDetailsService(x509UserDetailsService());
+                if (isAttlsEnabled) {
+                    http.addFilterBefore(new AttlsFilter(), X509AuthenticationFilter.class);
+                }
+            }
+        }
+
+        private UserDetailsService x509UserDetailsService() {
+            return username -> new User(username, "", Collections.emptyList());
+        }
     }
 
-    @Override
-    public void configure(WebSecurity web) {
-        // skip security filters matchers
-        String[] noSecurityAntMatchers = {
-            "/",
-            "/static/**",
-            "/favicon.ico",
-            "/api-doc"
-        };
-
-        web.ignoring().antMatchers(noSecurityAntMatchers);
+    /**
+     * Default filter chain to protect all routes with MF credentials.
+     */
+    @Configuration
+    public class FilterChainBasicAuthOrTokenAllEndpoints extends WebSecurityConfigurerAdapter {
+
+        @Override
+        protected void configure(AuthenticationManagerBuilder auth) {
+            auth.authenticationProvider(gatewayLoginProvider);
+            auth.authenticationProvider(gatewayTokenProvider);
+        }
+
+        @Override
+        public void configure(WebSecurity web) {
```

Comment:
Web security needs to be configured only once in one filterchain. Inspect it with spring security debug and you should see the evaluations are doubled.

Reply:
Fixed